### ISSUE TITLE
Eclipse XText/XTend for Java code generation

### ISSUE THREAD

**Issue by mohanr** [digital-asset/lib-cdm-event-specification-module#37]
A general question about your Java code generation proposal which may already exist. Have you considered Eclipse XText as the code generation engine ? DAML already seems to be like the XText DSL if I am not wrong. I am only familiar with basic Eclipse XText/XTend. I believe we don't need the Eclipse editor to do this. Just the Maven build file and dependencies.

**Comment by GeorgSchneider** [comment]
I don't think we would introduce another intermediate representation from DAML. If anything we would directly generate bytecode from DAML-LF. But we haven't pursued this approach further.

**Comment by mohanr** [comment]
Bytecode ? If you mean an ANTLR grammar like this https://github.com/antlr/grammars-v4/tree/master/haskell then it would be the similar for Java.

**Comment by GeorgSchneider** [comment]
No, I meant generate Java bytecode directly from DAML-LF, which is our intermediate representation
